### ISSUE TITLE
Discord Not Detected

### ISSUE THREAD

**Issue by TzZek** [jerwuqu/notificue#2]
Haven't found any other programs that don't work , only [Discord](https://discordapp.com/).

**Comment by jerwuqu** [comment]
Hello! Thanks for reporting this issue!

This project is a little on hold at the moment and when I start again it's going to be rewritten on [another branch](https://github.com/jerwuqu/notificue/tree/rewrite). I hope to get started on it soon. Having other use the project sure is a good motivator!

In the mean time, feel free to report any findings regarding the issue.

**Comment by TzZek** [comment]
Sounds good to me, actually was recommeneded this from a couple of other developers! Does exactly what we need it to and excited to see a new branch!

**Comment by JerwuQu** [comment]
Gave it a quick look and I think Discord is using Windows' new `ToastNotification` API which notificue doesn't support yet (notificue only supports the old Windows notification API). I'll look into supporting this in the rewritten version.

**Comment by JerwuQu** [comment]
So after researching a bit more I can solidly confirm that the Windows ecosystem is relatively fucked. The (with Windows 10) new ToastNotification (which can have buttons, dropdowns, or practically anything UWP has) is way too complex to simply replace or convert into text. The goal of this project is to be a minimalist replacement and as such, the exact opposite of that. I don't want arbitrary GUI elements in my notifications, not as a user nor a developer. 

The best I can do is to replace notifications for the old (and in some sense, better) API and let the new UWP/ToastNotification ones pass through without touching them, somewhat similar to how it's been up until now. The downside with this is that there will now be two different kinds of notifications of the users' system: one which the user likes and one which they does not. There _might_ be some way of tricking applications into thinking we're running Windows 7 and as such use the old notification API instead of the new one, which would be great, but I wouldn't bet on it. 

This is kind of a let-down as this is the project I use myself the most, so if a fellow developer sees this rant and has a solution that doesn't involve displaying the UWP stuff as-is (which I do not want), please do tell.

**Comment by TzZek** [comment]
> 
> 
> So after researching a bit more I can solidly confirm that the Windows ecosystem is relatively fucked. The (with Windows 10) new ToastNotification (which can have buttons, dropdowns, or practically anything UWP has) is way too complex to simply replace or convert into text. The goal of this project is to be a minimalist replacement and as such, the exact opposite of that. I don't want arbitrary GUI elements in my notifications, not as a user nor a developer.
> 
> The best I can do is to replace notifications for the old (and in some sense, better) API and let the new UWP/ToastNotification ones pass through without touching them, somewhat similar to how it's been up until now. The downside with this is that there will now be two different kinds of notifications of the users' system: one which the user likes and one which they does not. There _might_ be some way of tricking applications into thinking we're running Windows 7 and as such use the old notification API instead of the new one, which would be great, but I wouldn't bet on it.
> 
> This is kind of a let-down as this is the project I use myself the most, so if a fellow developer sees this rant and has a solution that doesn't involve displaying the UWP stuff as-is (which I do not want), please do tell.

That's unfortunate to hear, gotta love Windows 10... thanks for looking into it. Love the work you've done so far, exactly what I was looking for very similar to "Dunst" (which i love).

**Comment by JerwuQu** [comment]
Yeah, I'm also a dunst fan. It was an inspiration when creating this 😄 

Gonna dip my toes into other projects while I take a rest from this. 
I started working on [a dmenu for Windows](https://github.com/JerwuQu/wlines) a couple of days ago. There aren't any scripts made for it yet though, so I hope to get a couple made in the near future.

**Comment by TzZek** [comment]
> 
> 
> Yeah, I'm also a dunst fan. It was an inspiration when creating this 😄
> 
> Gonna dip my toes into other projects while I take a rest from this.
> I started working on [a dmenu for Windows](https://github.com/JerwuQu/wlines) a couple of days ago. There aren't any scripts made for it yet though, so I hope to get a couple made in the near future.

Well alright thanks, hopefully in the future something will be possible w/ Discord or perhaps I'll have to see if I can find something similar to [dLine](https://github.com/NatTupper/dline) on Windows. 🤔
